src/WebForms/UI/Control.cs

```diff
@@ -1016,15 +1027,14 @@ public string GetRouteUrl(RouteValueDictionary routeParameters)
         Justification = "Consistent with other URL properties in ASP.NET.")]
     public string GetRouteUrl(string routeName, RouteValueDictionary routeParameters)
     {
-        //TODO fix https://github.com/twsouthwick/systemweb-adapters-ui/issues/21
-        /*VirtualPathData data = RouteCollection.GetVirtualPath(Context.Request.RequestContext, routeName, routeParameters);
+        var requestContext = HttpContextHandlerExtensions.GetRequestContext(Context.AsAspNetCore());
```

Comment:
nit: this is an extension method right? can't you do `Context.AsAspNetCore().GetRequestContext()`?